fedimint-server/src/net/peers.rs

```diff
@@ -78,6 +80,52 @@ struct PeerConnectionStateMachine<M> {
     state: PeerConnectionState<M>,
 }
 
+struct PeerStatusQuery {
+    response: oneshot::Sender<PeerStatus>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum PeerStatus {
+    Connected,
+    Disconnected,
+}
+
+type PeerStatusChannelSender = Sender<PeerStatusQuery>;
+type PeerStatusChannelReceiver = Receiver<PeerStatusQuery>;
+
+#[derive(Clone)]
+pub struct PeerStatusClient(HashMap<PeerId, PeerStatusChannelSender>);
```

Comment:
Just to make it more self-explanatory I'd make it a :+1: 

```rust
struct PeerStatusClient {
  request_senders : HashMap<PeerId, PeerStatusChannelSender>,
}
```

BTW. The biggest problem with this rpc call approach (vs. just shared Arc<some_atomic_counters>` is that now there's bunch of extra code, and people trying to understand this code might get confused by the word `Client`, etc.

Please add a decently detailed comment that explains what is it for and roughly how it works.

Reply:
There is no client anymore :eyes: